src/background/feeder/e2e-session.feeder.ts

```diff
@@ -0,0 +1,194 @@
+import { nanoid } from 'nanoid'
+import { bzzProtocolToFakeUrl } from '../../contentscript/swarm-library/bzz-link'
+import { fakeUrl } from '../../utils/fake-url'
+import { appendSwarmSessionIdToUrl } from '../../utils/swarm-session-id'
+import { DappSessionManager } from '../dapp-session.manager'
+
+enum Action {
+  REGISTER = 'register',
+  LOCAL_STORAGE_GET = 'localStorage.getItem',
+  LOCAL_STORAGE_SET = 'localStorage.setItem',
+  BZZ_LINK_PROTOCOL_TO_FAKE_URL = 'bzzLink.bzzProtocolToFakeUrl',
+  BZZ_LINK_LINK_URL_TO_FAKE_URL = 'bzzLink.bzzLinkUrlToFakeUrl',
+  BZZ_LINK_URL_TO_FAKE_URL = 'bzzLink.urlToFakeUrl',
+  WEB2_HELPER_FAKE_BEE_API_ADDRESS = 'web2Helper.fakeBeeApiAddress',
+  WEB2_HELPER_FAKE_BZZ_ADDRESS = 'web2Helper.fakeBzzAddress',
+}
+
+interface Request<A extends Action, P> {
+  action: A
+  sessionId: string
+  parameters: P
+}
+
+interface Response {
+  data?: any
+  error?: string
+}
+
+type RegisterRequest = Request<Action.REGISTER, void>
+type LocalStorageGetRequest = Request<Action.LOCAL_STORAGE_GET, { name: string }>
+type LocalStorageSetRequest = Request<Action.LOCAL_STORAGE_SET, { name: string; value: unknown }>
+type BzzLinkProtocolToFakeUrlRequest = Request<Action.BZZ_LINK_PROTOCOL_TO_FAKE_URL, { url: string; newPage: boolean }>
+type BzzLinkLinkUrlToFakeUrlRequest = Request<
+  Action.BZZ_LINK_LINK_URL_TO_FAKE_URL,
+  { bzzLinkUrl: string; newPage: boolean }
+>
+type BzzLinkUrlToFakeUrlRequest = Request<Action.BZZ_LINK_URL_TO_FAKE_URL, { url: string; newPage: boolean }>
+type Web2HelperFakeBeeApiAddressRequesst = Request<Action.WEB2_HELPER_FAKE_BEE_API_ADDRESS, void>
+type Web2HelprFakeBzzAddressRequest = Request<Action.WEB2_HELPER_FAKE_BZZ_ADDRESS, { reference: string }>
+
+type RequestType =
+  | RegisterRequest
+  | LocalStorageGetRequest
+  | LocalStorageSetRequest
+  | BzzLinkProtocolToFakeUrlRequest
+  | BzzLinkLinkUrlToFakeUrlRequest
+  | BzzLinkUrlToFakeUrlRequest
+  | Web2HelperFakeBeeApiAddressRequesst
+  | Web2HelprFakeBzzAddressRequest
+
+export class E2ESessionFeeder {
+  constructor(private manager: DappSessionManager) {
+    this.serveEvents()
+  }
+
+  serveEvents(): void {
+    console.log('Register E2ESessionFeeder event listeners...')
+
+    // register dapp session id
+    chrome.runtime.onMessageExternal.addListener(
+      async (
+        request: RequestType,
+        sender: chrome.runtime.MessageSender,
+        sendResponse: (response?: Response) => void,
+      ) => {
+        const { action } = request || {}
+        const senderId = sender.id as string
+        const response: Response = {}
+
+        try {
+          if (!senderId) {
+            throw new Error('Invalid extension ID')
+          }
```

Comment:
move this snippet before the `try...catch` block so you don't have to assert `senderId`
```suggestion
          if (!senderId) {
            throw new Error('E2E Session Feeder: Extension ID is undefined.')
          }
```

Reply:
I intentionally left this if inside the try/catch block, because that way there is no need to call the 'handleResponse' function on two places. So whether there was an error or not, the 'handleResponse' will be called after the try/catch.